weetext.py

```diff
@@ -77,21 +80,28 @@
     "key_dir" : "/cryptkey",
     "cipher" : "aes-256-cbc",
     "message_indicator" : "(enc) ",
+    "api_key" : "" # google voice api key. TODO find dynamically?
 }
 
 conversation_map = {}
 number_map = {}
 conv = ''
 
 class Conversation(object):
-    def __init__(self, conv_id, number, messages):
-        self.conv_id = conv_id
+    def __init__(self, number, messages):
         self.number = number
-        self.messages = messages
+        self.__messages = OrderedDict([(msg["msg_id"], msg) for msg in messages])
```

Comment:
I left the format of the message item basically as it was in the original code. I'm storing them in an ordered dict so I can take the difference between the left and right hand side conversations during an update. The length of the conversation messages are set by the new API so the old system will always just create a copy when one is looking for the latest messages only.